Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -78,6 +78,19 @@ def edit
       end.presence
     end
 
+    unless Homebrew::EnvConfig.no_env_hints?
+      args.named.each do |name|
+        next if HOMEBREW_TAP_FORMULA_REGEX.match?(name)
+        next if name =~ HOMEBREW_TAP_CASK_REGEX && name !~ HOMEBREW_CASK_TAP_CASK_REGEX
+
+        opoo <<~EOS
+          Unless `HOMEBREW_NO_INSTALL_FROM_API` is set when running
+          `brew install`, it will ignore your locally edited formula.
+        EOS
```

Comment:
Yea, that would be ideal, I think. It's possible, just a bit ugly.

Reply:
Agree that would be ideal, but seems difficult since [looks like](https://github.com/Homebrew/brew/blob/1cf53b4281e7bc22a8d7fc915fc6ba4576f8ef88/Library/Homebrew/brew.sh#L807) it's always already automatically set when running `edit` as @apainintheneck mentioned.  
I can't think of a way to do this without editing stuff outside `edit.rb` and it feels a bit too much just for a warning, so I'll refrain from doing it

Reply:
Oh well, thank for looking into it.

Reply:
Once https://github.com/Homebrew/brew/pull/14613 is merged, you should be able to use `Homebrew::EnvConfig.no_install_from_api_set_automatically?` to check accomplish this